Description: patch to run multiple instances in (wmaker) dock
 When starting multiple instances of wmfire (one for cpu, one for net,
 etc) the window manager cannot see any difference between the instances.
 So it will randomly place it in the dock.
 .
 I have patched wmfire to set the X resource name by hand (option -r) so the
 window manager can see the difference. (It fixed my problem with wmaker)
 .
 patch included.
 .
 greetings,
 Christof
Author: Christof Douma <bugs@wizztick.net>
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=228357
Last-Update: 2015-01-15

--- a/src/wmfire.c
+++ b/src/wmfire.c
@@ -134,6 +134,7 @@
 int cpu_id = 0;
 int cpu_nice = 1;
 char net_dev[16] = "ppp0";
+char resource_name[64] = "wmfire";
 int net_spd = 0;
 char *file_name = NULL;
 int file_max = 100;
@@ -569,12 +570,12 @@
 
 	attr.width = 64;
 	attr.height = 64;
-	attr.title = "wmfire";
+	attr.title = resource_name;
 	attr.event_mask = MASK;
 	attr.wclass = GDK_INPUT_OUTPUT;
 	attr.visual = gdk_visual_get_system();
 	attr.colormap = gdk_colormap_get_system();
-	attr.wmclass_name = "wmfire";
+	attr.wmclass_name = resource_name;
 	attr.wmclass_class = "wmfire";
 	attr.window_type = GDK_WINDOW_TOPLEVEL;
 
@@ -653,7 +654,7 @@
 	bm.flame = fire[cmap].data;
 
 	/* Parse command options */
-	while ((i = getopt(argc, argv, "c:mni:s:xF:H:L:pf:lbhg:yS:")) != -1) {
+	while ((i = getopt(argc, argv, "c:mni:r:s:xF:H:L:pf:lbhg:yS:")) != -1) {
 		switch (i) {
 		case 'S':
 			if (optarg)
@@ -687,6 +688,10 @@
 			if (optarg)
 				strncpy(net_dev, optarg, sizeof (net_dev));
 			break;
+		case 'r':
+			if(optarg)
+				strcpy(resource_name, optarg);
+			break;
 		case 's':
 			if (optarg)
 				net_spd = atoi(optarg);
@@ -767,5 +772,6 @@
 		fprintf(stderr, "%d:%s ", i + 1, fire[i].text);
 	fprintf(stderr, "\n\t-l\t\t\tlock flame colour and monitor\n");
 	fprintf(stderr, "\t-b\t\t\tactivate broken window manager fix\n");
+	fprintf(stderr, "\t-r [...]\t\tchange resource name (default:%s)\n", resource_name);
 	fprintf(stderr, "\t-h\t\t\tprints this help\n");
 }
--- a/wmfire.1
+++ b/wmfire.1
@@ -98,6 +98,10 @@
 .B -p
 Fire effect only (no monitoring)
 .TP
+.B -r
+.RI [ ... ]
+Change resource name (default:wmfire)
+.TP
 .B -s [...][K|M]
 Change the network speed - kilobytes or megabytes
 .br
